File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/solution.cpp
#include <string>
#include <iostream>

class Solution {
public:
    long long concatenateNonZeroDigitsAndMultiplyBySum(int n) {
        if (n == 0) return 0;
        
        std::string s = std::to_string(n);
        std::string x_str = "";
        
        // Filter out non-zero digits
        for (char c : s) {
            if (c != '0') {
                x_str += c;
            }
        }
        
        // Handle edge case where no non-zero digits are found
        if (x_str.empty()) return 0;
        
        long long x = std::stoll(x_str);
        long long digit_sum = 0;
        
        // Calculate sum of digits of x
        for (char c : x_str) {
            digit_sum += (c - '0');
        }
        
        return x * digit_sum;
    }
};

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/solution.py
class Solution:
    def concatenateNonZeroDigitsAndMultiplyBySum(self, n: int) -> int:
        # Convert n to string to easily iterate over digits
        s = str(n)
        
        # Extract non-zero digits
        x_str = "".join([digit for digit in s if digit != '0'])
        
        # If no non-zero digits, result is 0
        if not x_str:
            return 0
            
        x = int(x_str)
        
        # Calculate sum of digits of x
        digit_sum = sum(int(digit) for digit in x_str)
        
        return x * digit_sum